src/app/(post)/result/[postId]/_components/CommentModalForm/CommentForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { useEffect } from 'react';
+
+import { useParams, useSearchParams } from 'next/navigation';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import { useGetCommentsAPI, useUpdateCommentAPI } from '@/src/apis/comments';
+import { useCreateReplyCommentAPI } from '@/src/apis/reply';
+import { useCreateComment } from '@/src/app/(post)/result/[postId]/hooks/useCreateComment';
+
+interface CommentFormProps {
+  isOpenModal: boolean;
+  onCloseModal: () => void;
+}
+
+export type CommentFormValues = {
+  comment: string;
+};
+
+const CommentForm = ({
+  isOpenModal,
+  onCloseModal: closeModal,
+}: CommentFormProps) => {
+  const { postId } = useParams() as { postId: string };
+  const method = useSearchParams().get('method') as 'create' | 'update' | null;
+  const target = useSearchParams().get('target') as 'root' | 'reply' | null;
+  const comments = useGetCommentsAPI(postId);
+  const commentId = useSearchParams().get('commentId');
```

Comment:
```suggestion
type MethodQueryStringType =  'create' | 'update' | null

const CommentForm = ({
  isOpenModal,
  onCloseModal: closeModal,
}: CommentFormProps) => {
  const { postId } = useParams() as { postId: string };
  const method = useSearchParams().get('method') as MethodQueryStringType 
  const target = useSearchParams().get('target') as 'root' | 'reply' | null;
  const comments = useGetCommentsAPI(postId);
  const commentId = useSearchParams().get('commentId');
```

간단한 comment는 그냥 아래서 단언으로 선언해주고 유니언타입을 오히려 선언해주는게 가독성이 더 좋아질 것 같아요